### ISSUE TITLE
Unable to load project requiring Maven 4.0.0-beta-5

### ISSUE THREAD

**Issue by gnodet** [grisha9/gmaven-plugin#20]
Trying to load mvnd with the following branch (with Maven 4.x poms)
https://github.com/apache/maven-mvnd/pull/1175

leads to the following error:
<img width="1057" alt="image" src="https://github.com/user-attachments/assets/c3eff25b-6e20-4b34-947e-0fb9e6e4e53e">

**Comment by grisha9** [comment]
Sorry... I cant reproduce it on my workstation OS Ubuntu
Can your please send more logs? Click on the mvnd:failed node and there will be a full maven log. send it.    
![image](https://github.com/user-attachments/assets/0f14efda-e0fd-48f6-9528-08e29b752bbb)

**Comment by grisha9** [comment]
For me it looks like this:   

https://github.com/user-attachments/assets/daa46197-64ac-4687-86cf-03cb85e93159

**Comment by gnodet** [comment]
That gives the following:
```
/Users/gnodet/.sdkman/candidates/java/21.0.4-tem/bin/java "-Dmaven.ext.class.path=/Users/gnodet/Library/Application Support/JetBrains/IntelliJIdea2025.1/plugins/gmaven/lib/maven-ext-event-handler-1.0-SNAPSHOT.jar" -Dgmaven.maven.home=/Users/gnodet/.m2/wrapper/dists/apache-maven-4.0.0-beta-5-bin/4mdhiqevrrl6utdgupl97k3b17/apache-maven-4.0.0-beta-5 -Dgmaven.plugin.annotation.paths.maven-compiler-plugin=annotationProcessorPaths -Dgmaven.plugins=org.apache.maven.plugins:maven-compiler-plugin;org.codehaus.mojo:build-helper-maven-plugin;dev.aspectj:aspectj-maven-plugin;org.codehaus.mojo:aspectj-maven-plugin;org.eclipse.tycho:tycho-compiler-plugin;org.codehaus.gmavenplus:gmavenplus-plugin;org.codehaus.gmaven:groovy-maven-plugin -Dgmaven.resolvedArtifactIds=aspectj-maven-plugin,aspectj-maven-plugin -Djava.awt.headless=true -Dfile.encoding=UTF-8 -Dsun.stdout.encoding=UTF-8 -Dsun.stderr.encoding=UTF-8 -classpath "/Users/gnodet/Library/Application Support/JetBrains/IntelliJIdea2025.1/plugins/gmaven/lib/server-1.0-SNAPSHOT.jar:/Users/gnodet/Library/Application Support/JetBrains/IntelliJIdea2025.1/plugins/gmaven/lib/server-api-1.0-SNAPSHOT.jar:/Users/gnodet/.m2/wrapper/dists/apache-maven-4.0.0-beta-5-bin/4mdhiqevrrl6utdgupl97k3b17/apache-maven-4.0.0-beta-5/boot/plexus-classworlds-2.8.0.jar:/Applications/IntelliJ IDEA 2025.1 EAP.app/Contents/lib/util_rt.jar:/Applications/IntelliJ IDEA 2025.1 EAP.app/Contents/plugins/maven-model/lib/maven-model.jar" ru.rzn.gmyasoedov.gmaven.server.RemoteGMavenServer
Port/ServicesPort/ID: 61254/64730/GMavenServerImpl81d89d9e

mvn ru.rzn.gmyasoedov:model-reader:1.2.3:resolve
```

**Comment by grisha9** [comment]
It looks like you have an old version of the plugin. What version of the IDEA and version GMaven do you have?

**Comment by gnodet** [comment]
Are you sure you've been using the branch I pointed at ?  Because it also works if I use `mvnd` master branch.

I'm using latest IntelliJ IDEA EAP (Build #IU-251.2067, built on October 25, 2024) and GMaven 233.30

**Comment by grisha9** [comment]
Got it. Thank you. I'll fix it today.

**Comment by grisha9** [comment]
Try 242.3.1 version from beta  https://plugins.jetbrains.com/plugin/22370-gmaven/versions/beta/630492?noRedirect=true
Updates in the main IDEA marketplace channel take several days

**Comment by grisha9** [comment]
Sorry. My stupid mistake. I was in a hurry and didn't check all the tests. 
Try 242.3.2 version from beta https://plugins.jetbrains.com/plugin/22370-gmaven/versions/beta/630539?noRedirect=true

**Comment by gnodet** [comment]
Same error:
```
mvnw -f /Users/gnodet/work/git/mvnd ru.rzn.gmyasoedov:maven-model-reader-plugin:0.2:resolve

/Users/gnodet/.m2/wrapper/dists/apache-maven-3.9.9-bin/4nf9hui3q3djbarqar9g711ggc/apache-maven-3.9.9/bin/mvn "-Dmaven.ext.class.path=\"/Users/gnodet/Library/Application Support/JetBrains/IntelliJIdea2025.1/plugins/gmaven/lib/gmaven-execution-event-spy-0.1.jar\"" -DresultAsTree=true -DresultFilePath=/Users/gnodet/work/git/mvnd/target/.gmaven.pom.json -DprocessingPluginGAIds=org.apache.maven.plugins:maven-compiler-plugin,org.codehaus.mojo:build-helper-maven-plugin,dev.aspectj:aspectj-maven-plugin,org.codehaus.mojo:aspectj-maven-plugin,org.eclipse.tycho:tycho-compiler-plugin,org.codehaus.gmavenplus:gmavenplus-plugin,org.codehaus.gmaven:groovy-maven-plugin -DresolvedPluginGAIds=dev.aspectj:aspectj-maven-plugin,org.codehaus.mojo:aspectj-maven-plugin -f /Users/gnodet/work/git/mvnd ru.rzn.gmyasoedov:maven-model-reader-plugin:0.2:resolve
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] Maven Daemon - Parent                                              [pom]
[INFO] Maven Daemon - Documentation Maven Plugin                 [maven-plugin]
[INFO] Maven Daemon - Native Library                                      [jar]
[INFO] Maven Daemon - Agent                                               [jar]
[INFO] Maven Daemon - Helper Agent                                        [jar]
[INFO] Maven Daemon - Common                                              [jar]
[INFO] Maven Daemon - Logging                                             [jar]
[INFO] Maven Daemon - Client                                              [jar]
[INFO] Maven Daemon - Daemon                                              [jar]
[INFO] Maven Daemon - Distribution                                        [pom]
[INFO] Maven Daemon - Integration Tests                                   [jar]
[WARNING] The POM for ru.rzn.gmyasoedov:maven-model-reader-plugin:jar:0.2 is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Maven Daemon - Parent 2.0.0-beta-2-SNAPSHOT:
[INFO] 
[INFO] Maven Daemon - Parent .............................. SKIPPED
[INFO] Maven Daemon - Documentation Maven Plugin .......... SKIPPED
[INFO] Maven Daemon - Native Library ...................... SKIPPED
[INFO] Maven Daemon - Agent ............................... SKIPPED
[INFO] Maven Daemon - Helper Agent ........................ SKIPPED
[INFO] Maven Daemon - Common .............................. SKIPPED
[INFO] Maven Daemon - Logging ............................. SKIPPED
[INFO] Maven Daemon - Client .............................. SKIPPED
[INFO] Maven Daemon - Daemon .............................. SKIPPED
[INFO] Maven Daemon - Distribution ........................ SKIPPED
[INFO] Maven Daemon - Integration Tests ................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.134 s
[INFO] Finished at: 2024-11-08T14:49:48+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin ru.rzn.gmyasoedov:maven-model-reader-plugin:0.2 or one of its dependencies could not be resolved:
[ERROR] 	ru.rzn.gmyasoedov:maven-model-reader-plugin:jar:0.2 was not found in https://repo.maven.apache.org/maven2/ during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginResolutionException
```

**Comment by gnodet** [comment]
Installing the plugin manually in the repo did work.

I think initially, gmaven was trying to use maven wrapper 3.9.9 instead of the version configured in `.mvn/wrapper`, but that went away when switching between maven / wrapper / daemon...

**Comment by grisha9** [comment]
Thank you for taking the time to help me find the problem with my plugin.

Problem in first run deal with: "mvn install:install-file" to local repo.
1) I'll try to reproduce this locally.
2) This is potentially a bottleneck. Best solution is to install my maven-plugin to maven central. I will work on it in with issue.

I will be glad if my plugin helps you in your work.

**Comment by grisha9** [comment]
I reproduced this locally. The problem is the space is not escaped correctly.
`"-Dmaven.ext.class.path=\"/IDEA Path/lib/gmaven-execution-event-spy-0.1.jar\""`
Fixed in 242.4 version https://plugins.jetbrains.com/plugin/22370-gmaven/versions/stable/630949?noRedirect=true

Created [issue](https://github.com/grisha9/gmaven-plugin/issues/21) for removed this logic "mvn install:install-file" and deploy my maven-plugin to maven central and migrate on it.

**Comment by grisha9** [comment]
fixed since 242.4
